ft-sdk/src/auth/mod.rs

```diff
@@ -73,29 +75,52 @@ pub fn session_providers() -> Vec<String> {
     todo!()
 }
 
+/// Fetches user data based on a given session cookie.
+///
+/// This function fetches user data based on a given session cookie if the
+/// session cookie is found and is valid and user data is found.
+/// If the session cookie not found, it returns `None`.
 #[cfg(feature = "field-extractors")]
 pub fn ud(
     cookie: ft_sdk::Cookie<SESSION_KEY>,
     conn: &mut ft_sdk::Connection,
 ) -> Result<Option<ft_sys::UserData>, UserDataError> {
-    if let Some(v) = ft_sys::env::var("DEBUG_LOGGED_IN".to_string()) {
-        let mut v = v.splitn(4, ' ');
-        return Ok(Some(ft_sys::UserData {
-            id: v.next().unwrap().parse().unwrap(),
-            identity: v.next().unwrap_or_default().to_string(),
-            name: v.next().map(|v| v.to_string()).unwrap_or_default(),
-            email: v.next().map(|v| v.to_string()).unwrap_or_default(),
-            verified_email: true,
-        }));
+    ft_sdk::println!("session cookie: {cookie}");
+    // Check if debug user data is available, return it if found.
+    if let Some(ud) = get_debug_ud() {
+        return Ok(Some(ud));
     }
 
-    ft_sdk::println!("sid: {cookie}");
-
-    let sid = match cookie.0 {
+    // Extract the session ID from the cookie.
+    let session_id = match cookie.0 {
         Some(v) => v,
         None => return Ok(None),
     };
 
+    ud_from_session_key(conn, &ft_sdk::auth::SessionID(session_id))
+}
+
+/// Fetches user data based on a given session id.
+///
+/// This function fetches user data based on a given session id if the
+/// session is valid and user data is found.
+/// If the session cookie not found, it returns `None`.
+#[cfg(feature = "field-extractors")]
+pub fn ud_from_session_key(
+    conn: &mut ft_sdk::Connection,
+    session_id: &ft_sdk::auth::SessionID,
+) -> Result<Option<ft_sys::UserData>, UserDataError> {
```

Comment:
I do not think we should have two method. Merge the two ud related methods please.

Reply:
Since we are not `session_id` passing in cookie but in `Authorization` header. We need to have other function to fetch ud from `Authorization` header with `FromRequest` implemented which I guess I am going to do.

Reply:
Change the original `ud()` method to accept `SessionID` instead of `Cookie<"session-id">`. Then you won't need this duplication.